### PR TITLE
Separate wh download

### PR DESCRIPTION
**Pull request by crazzysun**
Now Instance.details doesn't have workflowHistory, we can download it from router.get_instance_workflowhistory

### CODE REVIEW

**Comment by crazzysun** [comment]
@storgashov @vasichkin please review

File: qubell/api/private/instance.py [hunk 1]
```diff
@@ -256,6 +256,8 @@ def json(self):
         # noinspection PyAttributeOutsideInit
         self.__last_read_time = time.time()
         self.__cached_json = self._router.get_instance(org_id=self.organizationId, instance_id=self.instanceId).json()
+        self.__cached_json['workflowsInfo'] = self._router.get_instance_workflowhistory(org_id=self.organizationId, instance_id=self.instanceId).json()
```
Comment (vasichkin) [inline qubell/api/private/instance.py]: This looks strange for me and could have side effects. Why should you modify cache?

**Comment by vasichkin** [comment]
So, there won't be field 'workflowHistory' in instance details, right?
There are lots of dependencies on this field in tests. (https://github.com/qubell/vermilion/search?l=Python&q=workflowHistory&utf8=%E2%9C%93)

**Comment by vasichkin** [comment]
I got it.
I suggest add this workaround [here](https://github.com/qubell/contrib-python-qubell-client/blob/master/qubell/api/private/instance.py#L214) and please comment it.
Also it must be backward-compatible.
Modifying cache is bad idea.

**Comment by storgashov** [comment]
@vasichkin, the main idea was to make `Instance` behaving exactly as before. Retrieving "workflowInfo" within the `Instance.json` and adding it into the cache method does the work. If we move this operation into the `getattr`, the `Instance` won't be backward compatible because there're places in tests which get access to the instance's JSON directly and then extracting required fields, including "workflowInfo", from that JSON. The `getattr` is not called in such the places at all.

File: qubell/api/private/instance.py [hunk 2]
```diff
@@ -214,8 +214,14 @@ def __getattr__(self, key):
...
             j = self.json()
-            old_api_value = j.get(key)
-            new_api_value = j.get('workflowsInfo', {}).get(key, False)
+            old_api_value = j.get('workflowsInfo', {}).get(key, False)
+
+            workflowHistory_len = len(j.get('workflowsInfo', {}).get("workflowHistory", {}))
+            scheduledWorkflows_len = len(j.get('workflowsInfo', {}).get("scheduledWorkflows", {}))
+            availableWorkflows_len = len(j.get('workflowsInfo', {}).get("availableWorkflows", {}))
+            new_workflowsInfo =  self._router.get_instance_workflowhistory(org_id=self.organizationId, instance_id=self.instanceId).json()
+            new_api_value = False if (workflowHistory_len + scheduledWorkflows_len + availableWorkflows_len > 0) else new_workflowsInfo.get(key, False)
```
Comment (vasichkin) [inline qubell/api/private/instance.py]: I'd like None instead of False, since it's default for get().

File: qubell/api/private/instance.py [hunk 3]
```diff
@@ -214,8 +214,14 @@ def __getattr__(self, key):
...
             log.debug('Getting instance workflow attribute: %s' % key)
             j = self.json()
-            old_api_value = j.get(key)
-            new_api_value = j.get('workflowsInfo', {}).get(key, False)
+            old_api_value = j.get('workflowsInfo', {}).get(key, False)
+
+            workflowHistory_len = len(j.get('workflowsInfo', {}).get("workflowHistory", {}))
+            scheduledWorkflows_len = len(j.get('workflowsInfo', {}).get("scheduledWorkflows", {}))
+            availableWorkflows_len = len(j.get('workflowsInfo', {}).get("availableWorkflows", {}))
+            new_workflowsInfo =  self._router.get_instance_workflowhistory(org_id=self.organizationId, instance_id=self.instanceId).json()
```
Comment (vasichkin) [inline qubell/api/private/instance.py]: So, will it raise on old platform version?

**Comment by vasichkin** [comment]
Looks good to me, but should be tested.
Move client submodule in your PR to vermilion and pass tests. When passed - could be merged.

**Pull request closed without merging**